### ISSUE TITLE
RSS / Rack Initialize: Fail fast and explicitly

### ISSUE THREAD

**Issue by rcgoodfellow** [oxidecomputer/omicron#3198]
When we send the rack-initialize call, before RSS sends out the signal for bootstrap agents to initialize the rack, there should be a preliminary check done first. In this check, RSS will contact each of the sleds it expects to be in the cluster on the bootstrap addresses provided in the rack initialize call. The RSS agent will ask each of the sled agents to verify that it's ready to initialize the rack. The individual sled/bootstrap agents then check all the preconditions for initializing the sled and report back to RSS. If there are failures, all of them should be reported back to the caller so they can take remedial action. This includes not being able to contact expected sleds. This should be a fast operation, on the order of a few seconds.

cc @andrewjstone

**Comment by andrewjstone** [comment]
This will be done automatically by the trust quorum, which will run before any other RSS initialization. Bootstrap addresses will get passed and shares will get distributed and written to the M.2s before any other setup is done. A timeout will be set, such that if it doesn't complete we will report the failure, debug, and eventually reset the rack and try again.

**Comment by askfongjojo** [comment]
@rcgoodfellow: Can we consider this closed now that trust quorum is in place?

**Comment by rcgoodfellow** [comment]
Having a trust quorum has us in a much better place for the fast part. I think there is still ground to cover for the explicit part. At any time during rack initialization, the operator should be able to ask for a rack initialization report that includes pending and/or failed initialization tasks. E.g. operators should be able to explicitly ask why is the rack not initialized yet and get an explicit answer.

**Comment by jgallagher** [comment]
Agreed with @rcgoodfellow - the only operartor-visible feedback is wicket's `Initializing` (with no further info). In addition to the internal-to-the-rack checks that trust quorum does, if RSS is stuck or doomed without external changes (to pull some possible problems out of a hat: our transceivers are failing to uplink, or we're stuck waiting on timesync because we're firewalled off from the NTP servers we're supposed to sync with), there's no way to know that without being logged into the support account and digging around.
